src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseAnalysisServiceImpl.java

```diff
@@ -183,15 +190,43 @@ public Analysis saveResults(Analysis analysis, File resultDir) {
 
         Analysis updated = null;
         try {
-            updated = update(analysis);
+            updated = update(analysis, resultDir);
         } catch (NotExistException e) {
             LOGGER.warn(ANALYSIS_IS_NOT_EXISTS_LOG, analysis.getId());
         }
 
         return updated;
     }
 
-    private Analysis update(Analysis analysis) throws NotExistException {
+    private AnalysisResultStatusDTO reEvaluateAnalysisStatus(AnalysisResultStatusDTO originalStatus, File resultDir){
```

Comment:
can you extract evaluate status logic in a separate class?

Reply:
+ add unit tests

Reply:
done